File: modulo1/src/ros2_ws/src/fulgor_cpp_examples/src/publisher.cpp
//Descripción: Nodo publicador que envía un mensaje del tipo String al tópico /información cada 2 seg.

#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/string.hpp"

using namespace std::chrono_literals;

//define una clase Publisher (crea un nodo) que hereda de rclcpp::Node
class Publisher: public rclcpp::Node
{
public:
    //constructor
    Publisher(): Node("publisher")
    {
        //creación del publisher
        publisher_ = this -> create_publisher<example_interfaces::msg::String>("informacion", 10);
        
        //creación del timer
        timer_ = this ->create_wall_timer(2.0s, std::bind(&Publisher::publishInfo, this));
        RCLCPP_INFO(this->get_logger(), "Nodo creado correctamente!"); 
    }
private:

    //función que ejecuta el timer
    void publishInfo()
    {
        //creación del msj y asignación del dato en el campo
        auto msg = example_interfaces::msg::String();
        msg.data = "Hello world!";

        //publicación en el tópico
        publisher_->publish(msg);
    }

    //Publisher guardado como SharedPtr para que viva mientras el nodo esté activo..
    rclcpp::Publisher<example_interfaces::msg::String>::SharedPtr publisher_;

    //Timer guardado como SharedPtr para evitar que se destruya 
    rclcpp::TimerBase::SharedPtr timer_;
};


//función principal
int main (int argc, char **argv)
{
    //inicializa ros 2, crea el nodo y lo mantiene activo
    rclcpp::init(argc, argv);
    auto node = std::make_shared<Publisher>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: modulo1/src/ros2_ws/src/fulgor_py_examples/fulgor_py_examples/publisher.py
#Descripción: Nodo publicador que envía un mensaje del tipo String al tópico "informacion" cada 2 seg.

#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.msg import String

#definición del nodo
class Publisher(Node):
   
    def __init__(self):
        
        #constructor del nodo
        super().__init__("publisher")

        #creación del publisher
        self.publisher_ = self.create_publisher(String, "informacion", 10)
        self.get_logger().info("Nodo publisher creado exitosamente!")

        #creación del timer
        self.timer_ = self.create_timer(2,self.publish_info_)

    def publish_info_(self):
        
        #creación del msj y asignación del dato en el campo
        msg = String ()
        msg.data = "hola mundo!"

        #publicación en el tópico
        self.publisher_.publish(msg)

#función principal
def main(args=None):
   
    #inicializa ros 2, crea el nodo y lo mantiene activo
    rclpy.init(args=args)
    node = Publisher()
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == "__main__":
    main()
